projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1943 @@
+	UART_TX("                                  <ampl> = amplitude in mV, 1 - 3 mV"
+		_EOS);
+	UART_TX("pulseduration <dur>            -Set duration for pulse test" _EOS);
+	UART_TX("                                  <dur> = duration in ms, 10 to 200"
+		_EOS);
+	UART_TX("readsensors                    -Read the ppb value of all sensors"
+		_EOS);
+	UART_TX("setupdaterate <time>           -Configure update rate for stream mode"
+		_EOS);
+	UART_TX("stopread                       -Stop sensor data reading" _EOS);
+	UART_TX("\r\n" _EOS);
+}
+
+/**
+ @brief Finds the next command line argument
+
+ @param args - pointer to the arguments on the command line.
+
+ @return pointer to the next argument.
+
+ **/
+uint8_t *FindArgv(uint8_t *args)
+{
+	uint8_t *p = args;
+	int fl = 0;
+
+	while (*p != 0) {
+		if ((*p == _SPC))
+			fl = 1;
+		else if (fl)
+			break;
+		p++;
+	}
+
+	return p;
+}
+
+/**
+ @brief Separates a command line argument
+
+ @param dst - pointer to a buffer where the argument will be copied
+ @param args - pointer to the current position of the command line .
+
+ @return none
+
+ **/
+void GetArgv(char *dst, uint8_t *args)
+{
+	uint8_t *s = args;
+	char *d = dst;
+
+	while (*s) {
+		if (*s == _SPC)
+			break;
+		*d++ = *s++;
+	}
+
+	*d = '\0';
+}
+
+/**
+ @brief Command line process function
+
+ @return none
+ **/
+void CmdProcess(void)
+{
+	cmdFunc func;
+
+	/* <ENTER> key was pressed */
+	/* Find needed function based on typed command */
+	func = FindCommand((char *) cmdInString);
+
+	/* Check if there is a valid command */
+	if (func) {
+		/* Call the desired function */
+		(*func)(&cmdInString[2]);
+
+		/* Check if there is no match for typed command */
+	} else if (cmdInString[0] != 0) { //equivalent to strlen = 0
+		/* Display a message for unknown command */
+		UART_TX("Unknown command!" _EOS);
+	}
+	/* Prepare for next <ENTER> */
+	CmdPrompt();
+}
+
+/**
+ @brief Command line prompt. Caller must verify that only the enabled board can run this.
+
+ @return int - UART status: UART_SUCCESS or error status
+ **/
+void CmdPrompt(void)
+{
+	static uint8_t count = 0;
+
+	/* Check first <ENTER> is pressed after reset */
+	if (count == 0) {
+		UART_TX("Welcome to CN0429!" _EOS);
+		UART_TX("Type <help> to see available commands..." _EOS);
+		count++;
+	}
+
+}
+
+/**
+ @brief Find available commands
+
+ @param cmd - command to search
+
+ @return cmdFunc - return the specific function for available command or NULL
+ for invalid command
+ **/
+cmdFunc FindCommand(char *cmd)
+{
+	cmdFunc func = NULL;
+	char *p = cmd;
+	int i = 0;
+	int f1 = 0;
+	int cmdlength = 0;
+
+	while (*p != 0) {
+		if (*p == _SPC)
+			f1 = 1;
+		else if (!f1)
+			cmdlength++;
+		p++;
+	}
+
+	while (CmdFun[i] != NULL) {
+		if (strncmp(cmd, CmdCommands[i], cmdlength) == 0) {
+			func = CmdFun[i];
+			break;
+		}
+		i++;
+	}
+
+	return func;
+}
+
+void DoNothing(uint8_t *args)
+{
+	// nothing to do here
+	// this is to avoid help printing each time enter is pressed
```

Comment:
This one is still necessary. The issue was that the zero length string when someone pressed enter with no command typed in would register as a match for the first command in the CmdCommands[] list. So if you pressed enter, the CmdHelp would run. This function is here so that it runs on an empty command instead of CmdHelp.